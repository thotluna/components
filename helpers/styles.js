import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
