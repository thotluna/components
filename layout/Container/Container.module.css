.container {
  width: 100%;
  max-width: 800px;
  margin: 0 auto;
}

.container.type-full {
  padding: 2rem;
}

.container.type-withoutup {
  padding: 0 2rem 2rem;
}

.container.is-playground {
  border: var(--border-width-thick) dashed var(--color-primary);
}
