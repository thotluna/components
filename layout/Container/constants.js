export const options = {
  types: ['full', 'withoutup'],
}
