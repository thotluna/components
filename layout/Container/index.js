export { default } from './Container'
export { options } from './constants'
export { default as styles } from './Container.module.css'
