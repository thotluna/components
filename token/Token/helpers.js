import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
