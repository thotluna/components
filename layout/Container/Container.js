import React from 'react'
import PropTypes from 'prop-types'
import styles from './Container.module.css'
import { getClasses } from '../../helpers/styles'
import { options } from './constants'

export const Container = ({ children, type, isPlayground }) => {
  const getStyles = getClasses(styles)({ type })
  return (
    <div
      className={getStyles('container', ['type'], {
        'is-playground': isPlayground,
      })}
    >
      {children}
    </div>
  )
}

Container.propTypes = {
  children: PropTypes.node,
  type: PropTypes.oneOf(options.types).isRequired,
  isPlayground: PropTypes.bool,
}

Container.defaultProps = { isPlayground: false }

export default Container
