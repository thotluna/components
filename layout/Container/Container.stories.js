/* eslint-disable react/no-unescaped-entities */
import React from 'react'
import Container, { options } from '.'
import { getOptionsArgTypes, getVariusTemplate } from '../../helpers/storybook'

const Template = (args) => <Container {...args} />
const VariusTemplage = getVariusTemplate(Container)

export default {
  title: 'Layout/Container',
  component: Container,
  args: {
    children: (
      <p style={{ color: '#ffffff', fontSize: '18px' }}>
        I can't remember anything Can't tell if this is true or dream Deep down
        inside I feel the scream This terrible silence stops in me Now that the
        war is through with me I'm waking up, I cannot see That there is not
        much left of me Nothing is real but pain now Hold my breath as I wish
        for death Oh please, God, wake me
      </p>
    ),
    isPlayground: true,
    type: 'full',
  },
  argTypes: {
    children: { control: 'text' },
    isPlayground: { control: 'boolean' },
    type: getOptionsArgTypes(options.types),
  },
}

export const Default = Template.bind({})

export const Types = VariusTemplage.bind({})
Types.args = { items: options.types.map((type) => ({ type })) }
