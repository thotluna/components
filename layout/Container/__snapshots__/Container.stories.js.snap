// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/Container Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container type-full is-playground"
  >
    <p
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "18px",
        }
      }
    >
      I can't remember anything Can't tell if this is true or dream Deep down inside I feel the scream This terrible silence stops in me Now that the war is through with me I'm waking up, I cannot see That there is not much left of me Nothing is real but pain now Hold my breath as I wish for death Oh please, God, wake me
    </p>
  </div>
</div>
`;

exports[`Storyshots Layout/Container Types 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container type-full is-playground"
  >
    <p
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "18px",
        }
      }
    >
      I can't remember anything Can't tell if this is true or dream Deep down inside I feel the scream This terrible silence stops in me Now that the war is through with me I'm waking up, I cannot see That there is not much left of me Nothing is real but pain now Hold my breath as I wish for death Oh please, God, wake me
    </p>
  </div>
  <div
    className="container type-withoutup is-playground"
  >
    <p
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "18px",
        }
      }
    >
      I can't remember anything Can't tell if this is true or dream Deep down inside I feel the scream This terrible silence stops in me Now that the war is through with me I'm waking up, I cannot see That there is not much left of me Nothing is real but pain now Hold my breath as I wish for death Oh please, God, wake me
    </p>
  </div>
</div>
`;
